### PR TITLE
chore: structure in `impl` + `api`

### PR DESCRIPTION
**Pull request by nikku**
## Description

Proposal to re-structure the job worker run-time in accordance with our `impl` / `api` package pattern.

### CODE REVIEW

File: runtime-job-worker/src/main/java/io/camunda/connector/runtime/jobworker/impl/outbound/feel/FeelEngineWrapper.java [hunk 1]
```diff
@@ -14,7 +14,7 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package io.camunda.connector.runtime.jobworker.feel;
+package io.camunda.connector.runtime.jobworker.impl.outbound.feel;
```
Comment (tmetzke) [inline runtime-job-worker/src/main/java/io/camunda/connector/runtime/jobworker/impl/outbound/feel/FeelEngineWrapper.java:17]: Looking at how Bernd worked with this, we will probably reuse the wrapper also for inbound. I think we can actually keep it without `outbound`. But in the end, it's not API, so wet could also change this later as well.

File: runtime-job-worker/src/main/java/io/camunda/connector/runtime/jobworker/impl/outbound/feel/FeelEngineWrapperException.java [hunk 2]
```diff
@@ -14,7 +14,7 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package io.camunda.connector.runtime.jobworker.feel;
+package io.camunda.connector.runtime.jobworker.impl.outbound.feel;
```
Comment (tmetzke) [inline runtime-job-worker/src/main/java/io/camunda/connector/runtime/jobworker/impl/outbound/feel/FeelEngineWrapperException.java:18]: Same as the wrapper class.

File: runtime-job-worker/src/main/java/io/camunda/connector/runtime/jobworker/api/outbound/ConnectorJobHandler.java [hunk 3]
```diff
@@ -15,16 +15,16 @@
  * limitations under the License.
  */
 
-package io.camunda.connector.runtime.jobworker.outbound;
+package io.camunda.connector.runtime.jobworker.api.outbound;
```
Comment (tmetzke) [inline runtime-job-worker/src/main/java/io/camunda/connector/runtime/jobworker/api/outbound/ConnectorJobHandler.java:18]: We need to move the class accordingly as well then.

**Review by tmetzke — COMMENTED** [review]
I like it, way more consistent 👍
The FEEL classes don't have to move necessarily, I think. And one of the classes needs to be moved according to the new package, if I'm not mistaken.

File: runtime-job-worker/src/main/java/io/camunda/connector/runtime/jobworker/api/outbound/ConnectorJobHandler.java (hunk 3 above)
Comment (nikku) [reply runtime-job-worker/src/main/java/io/camunda/connector/runtime/jobworker/api/outbound/ConnectorJobHandler.java:18]: I'm honestly not sure what IDEA is doing here. Why do I use an IDE in the first place if I cannot rely on it to "MOVE" things properly. :see_no_evil:

File: runtime-job-worker/src/main/java/io/camunda/connector/runtime/jobworker/impl/outbound/feel/FeelEngineWrapper.java (hunk 1 above)
Comment (nikku) [reply runtime-job-worker/src/main/java/io/camunda/connector/runtime/jobworker/impl/outbound/feel/FeelEngineWrapper.java:17]: Agreed + we can likely make it a utility in connectors generally at some point.

This is missing in the Camunda stack: A proper Java wrapper around feel-scala.

File: runtime-job-worker/src/main/java/io/camunda/connector/runtime/jobworker/api/outbound/ConnectorJobHandler.java (hunk 3 above)
Comment (nikku) [reply runtime-job-worker/src/main/java/io/camunda/connector/runtime/jobworker/api/outbound/ConnectorJobHandler.java:18]: Class moved.

File: runtime-job-worker/src/main/java/io/camunda/connector/runtime/jobworker/impl/outbound/feel/FeelEngineWrapper.java (hunk 1 above)
Comment (nikku) [reply runtime-job-worker/src/main/java/io/camunda/connector/runtime/jobworker/impl/outbound/feel/FeelEngineWrapper.java]: Moved back out of outbound. We can move these folks around anytime in the future again.

File: runtime-job-worker/src/main/java/io/camunda/connector/runtime/jobworker/impl/outbound/feel/FeelEngineWrapperException.java (hunk 2 above)
Comment (nikku) [reply runtime-job-worker/src/main/java/io/camunda/connector/runtime/jobworker/impl/outbound/feel/FeelEngineWrapperException.java]: Moved.

**Review by tmetzke — APPROVED** [review]
Nice 👍

**Comment by tmetzke** [comment]
@nikku, can you add this to the update guide in https://github.com/camunda/camunda-platform-docs/pull/1255 as well?

**Comment by nikku** [comment]
Done via https://github.com/camunda/camunda-platform-docs/pull/1255/commits/3b0763da438c8c86163bff75827fa38871453077, please have a look.

**Pull request merged**